### DISCUSSION TITLE
Docker image creation taking too long to finish

### DISCUSSION THREAD

**Discussion by ghost** [community/community#25112] [Actions]
<p>Hello.</p>
<p>I just followed <a href="https://lab.github.com/githubtraining/github-actions:-publish-to-github-packages" rel="nofollow noopener">this course</a> on how to publish docker images as packages from GitHub Actions but I’m having some problems with the job taking too long to finish when I try it out in my project.</p>
<p>The goal for me is to build a simple discord bot written in node and having watchtower on vultr vps update a docker container when a new image is available.</p>
<p>My docker image creation in my workflow currently takes 3m 22s to finish and I’m wondering if I’m doing something wrong here because it took just about 9 seconds to build the image when I took the course. So I’m asking if there is something I can do to make this process go a little bit faster because I think it takes too long.</p>
<p>My workflow:</p>
<pre><code>name: CI

on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master

jobs:
  build:
    runs-on: ubuntu-18.04
    steps:
    - uses: actions/checkout@v2

    - name: Install dependencies
      run: npm install

    - name: Run eslint
      run: npx eslint .

    - name: Create Docker image
      uses: mattdavis0351/actions/docker-gpr@v1.3.0
      with:
        repo-token: ${{ secrets.GITHUB_TOKEN }}
        image-name: twitter-discord-bot
</code></pre>
<p>My Dockerfile:</p>
<pre><code>FROM node:12
WORKDIR /usr/src/app
COPY package*.json ./
RUN npm ci --only=production
COPY . .
CMD ["npm", "start"]
</code></pre>
<p>When I check the output I see that Github tries to start my docker image by doing “npm start”. I meant this only to be when the docker container is executing not for actually building the docker image.</p>
<pre><code>Step 6/6 : CMD ["npm", "start"]
</code></pre>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/hornta">@hornta</a> ,</p>
<p>The docker image build time is related to the Dockerfile how you defined. It looks correct for your build time.</p>
<p>Compare with the course, image ‘node:12’ is used and added some extra cmd (eg: npm…), it will take a little more time than the course.</p>
<p>In addition, if you trigger the workflow again, you will find it takes less time due to docker layer exists.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2b988a9ad4b16c8d3411f959fc69feb2271a6989.png" data-download-href="/uploads/short-url/6dFlGliA6NaPeZhS4CbVvILucQ1.png?dl=1" title="dockerbuild.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181088981-c7678425-b9d9-4db8-85b4-3bb8e4f53e78.png" alt="dockerbuild.png" data-base62-sha1="6dFlGliA6NaPeZhS4CbVvILucQ1" width="690" height="338" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2b988a9ad4b16c8d3411f959fc69feb2271a6989_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">dockerbuild.png</span><span class="informations">990×486 30.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
